### ISSUE TITLE
Can we use machine learning tools with MATSim ?

### ISSUE THREAD

**Issue by TaghreedAlghamdi** [matsim-org/matsim-code-examples#517]
Hello!
I'm here looking for some answers for a possible contribution with MATSim. 
I'm interested in using MATSim with  machine learning tools (different spatiotemporal models to do analysis and prediction).

I need some clarification please regarding whether it would be possible to use a machine learning tools with MATSim such as Bayesian models?

I want to use MATSim with traffic data, and the contribution will be using Bayesian model as component which will be built in java and plugged into the MATSim environment as a package. 

The traffic data will include the coordinates ( latitude, longitude), time (HH:MM:SS) and traffic volume.

In MATSim, I want to create events profile (rainfall, snowfall) and model the parameters of these events to show the effect of the event on the traffic status when it happened( for example, the traffic volume will increase with specific percent at location A at time 6pm when the snowfall event occurs.) 

The Bayesian model will  plugged into MATSim to describe and predict the traffic status in the future when one of these events happen.

I was looking for research that used MATSim with machine learning tools  but I wan't lucky and I'm not sure if MATSim framework will be flexible to include machine learning tools . 

I hope I can find answers here , and any other sources that could help, will be much appreciated.
Thank you!

**Comment by JWJoubert** [comment]
I do not think that is quite the purpose of MATSim. Yes, it is used to evaluate interventions and policy, but only to the extent that you can capture the behavioural response in the model _a priori_. From your description, it seems you want to apply Bayesian models on _historical data_ for which you already know the state of the dependent variable, like the weather, and then learn/train the model. 

You may train a Bayesian network outside of MATSim and then embed it inside as a possible behavioural model (as opposed to, for example, a discrete choice model). Maybe others have different views on this.

**Comment by TaghreedAlghamdi** [comment]
Thank you for your feedback! And yes, I'm thinking to use Bayesian to implement and train based on historical data and then embed it into MATSim as a classifier, and I was looking for any MATSim package that has similar idea. 
I'll look into the Discrete Mode Choice extension, and I'll be grateful for more feedback. 

Thank you!
